Simplified/Settings/NYPLSettingsAccountsList.swift

```diff
@@ -199,23 +194,55 @@
       // Neither library is special so we just go alphabetically.
       return a.name.localizedCaseInsensitiveCompare(b.name) == .orderedAscending
     }
-
-    for userAccount in sortedLibraryAccounts {
-      if (!userAddedSecondaryAccounts.contains(userAccount.uuid) && userAccount.uuid != manager.currentAccount?.uuid) {
-        alert.addAction(UIAlertAction(title: userAccount.name,
-          style: .default,
-          handler: { action in
-            self.userAddedSecondaryAccounts.append(userAccount.uuid)
-            self.updateSettingsAccountList()
-            self.updateNavBar()
-            self.tableView.reloadData()
-        }))
-      }
+    
+    let finderBusinessLogic = NYPLLibraryFinderBusinessLogic(userAccounts: userLibraryAccounts, newLibraryAccounts: sortedLibraryAccounts)
+    let finderVC = NYPLLibraryFinderViewController(dataProvider: finderBusinessLogic) { [weak self] account in
+      self?.userAddedSecondaryAccounts.append(account.uuid)
+      self?.updateSettingsAccountList()
+      self?.updateNavBar()
+      self?.tableView.reloadData()
+      self?.navigationController?.popViewController(animated: true)
     }
-
-    alert.addAction(UIAlertAction(title: NSLocalizedString("Cancel", comment: "Cancel button title"), style: .cancel, handler:nil))
+    self.navigationController?.pushViewController(finderVC, animated: true)
     
-    self.present(alert, animated: true, completion: nil)
+//    let alert = UIAlertController(title: NSLocalizedString(
```

Comment:
is this the old code? I'm guessing you want to leave it here for now?

Reply:
Yeah, I will do some refactoring when everything is settled